gnomad_qc/v5/sample_qc/create_sample_qc_metadata_ht.py

```diff
@@ -0,0 +1,571 @@
+    meta_ht = meta_ht.annotate(
+        relatedness_inference=hl.struct(
+            release_relatedness_filters=relatedness_filters_ht[
+                meta_ht.key
+            ].release_relatedness_filters,
+            relationships=relatedness_inference_ht[meta_ht.key].relationships,
+        )
+    )
+
+    return meta_ht
+
+
+def get_relatedness_dict_ht(
+    ht: hl.Table,
+    filter_exprs: Dict[str, hl.expr.BooleanExpression] = None,
+) -> hl.Table:
+    """
+    Parse relatedness Table to get every relationship (except UNRELATED) per sample.
+
+    Return Table keyed by sample with all sample relationships in dictionary where the
+    key is the relationship and the value is a set of all samples with that
+    relationship to the given sample.
+
+    :param ht: Table with inferred relationship information. Keyed by sample pair (i, j).
+    :param filter_exprs: Optional dictionary of filter expressions to apply to `ht`
+        before creating the 'relationships' annotations. Keyed by the postfix to add to
+        'relationships' as the annotation label, and with boolean expressions as the
+        values. By default, no additional filtering is applied, and a single
+        'relationships' annotation is created.
+    :return: Table keyed by sample (s) with all relationships annotated as a dict.
+    """
+    if filter_exprs is None:
+        filter_exprs = {"": True}
+
+    # Add annotations for the relationship of each item in filter_expr.
+    ht = ht.annotate(
+        **{
+            f"_relationship{label}": hl.or_missing(expr, ht.relationship)
+            for label, expr in filter_exprs.items()
+        }
+    )
+    relationship_labels = [f"_relationship{label}" for label in filter_exprs]
+
+    # Filter to only pairs that are related (second-degree or closer).
+    ht = ht.filter(ht.relationship != UNRELATED)
+
+    # Build a Table of relationships duplicating the info for each pair. Pair (i, j)
+    # will have two rows, one for (s: i.s, pair: j.s) and one for (s: j.s, pair: i.s).
+    ht = ht.select(*relationship_labels, s=ht.i.s, pair=ht.j.s).union(
+        ht.select(*relationship_labels, s=ht.j.s, pair=ht.i.s)
+    )
+
+    # Group the Table by the sample name (s) and aggregate to get a relationships
+    # dictionary per sample. The dictionary is built by grouping the relationship
+    # annotation (becomes the key) and collecting the set of all samples (value) with
+    # each relationship. Each item in filter_expr has its own annotation.
+    ht = ht.group_by(ht.s).aggregate(
+        **{
+            f"relationships{label}": hl.agg.filter(
+                hl.is_defined(ht[f"_relationship{label}"]),
+                hl.agg.group_by(
+                    ht[f"_relationship{label}"], hl.agg.collect_as_set(ht.pair)
+                ),
+            )
+            for label in filter_exprs
+        }
+    )
+
+    return ht
+
+
+def get_relationship_filter_expr(
```

Comment:
basic question, could you import this from v4? or should we add this function to gnomad_methods?